wallet-sdk/src/main/kotlin/org/stellar/walletsdk/Anchor.kt

```diff
@@ -245,4 +243,73 @@ class Anchor(
         .transactions
     }
   }
+
+  /**
+   * Get account transactions for specified asset. Optional field implementation depends on anchor.
+   *
+   * @param assetCode asset's code
+   * @param authToken auth token of the account authenticated with the anchor
+   * @param toml Anchor's stellar.toml file containing `CURRENCIES` list of supported assets
+   * @param limit optional how many transactions to fetch
+   * @param pagingId optional return transactions prior to this ID
+   * @param noOlderThan optional return transactions starting on or after this date and time
+   * @param lang optional language code specified using
+   * [RFC 4646](https://www.rfc-editor.org/rfc/rfc4646), default is `en`
+   *
+   * @return a list of formatted operations
+   *
+   * @throws [NetworkRequestFailedException] if network request fails
+   */
+  suspend fun getHistory(
+    assetCode: String,
+    authToken: String,
+    toml: Map<String, Any>,
+    limit: Int? = null,
+    pagingId: String? = null,
+    noOlderThan: String? = null,
+    lang: String? = null
+  ): List<WalletOperation<AnchorTransaction>> {
+    val anchorCurrency =
+      ((toml as HashMap)["CURRENCIES"] as List<*>).filterIsInstance<HashMap<*, *>>().find {
+        it["code"] == assetCode
+      }
+      // TODO: custom exception
+      ?: throw Exception("Anchor does not support $assetCode asset")
+    val asset = Asset.create("$assetCode:$anchorCurrency[\"issuer\"]")
+
+    val transferServerEndpoint = toml[StellarTomlField.TRANSFER_SERVER_SEP0024.text].toString()
+    val endpointHttpUrl = transferServerEndpoint.toHttpUrl()
+    val endpointUrl = HttpUrl.Builder().scheme("https").host(endpointHttpUrl.host)
+
+    // Add path segments, if there are any
+    endpointHttpUrl.pathSegments.forEach { endpointUrl.addPathSegment(it) }
+
+    // Add transactions path segment
+    endpointUrl.addPathSegment("transactions")
+
+    // Add query params
+    val queryParams = mutableMapOf<String, String>()
+    queryParams["asset_code"] = assetCode
+    queryParams["limit"] = limit?.toString() ?: ""
```

Comment:
[nit] I think you should not set query params at all if the field is null (limit = "" and limit = null are different, so it may have a different behavior)

Reply:
I thought it would be easier to read if all of them are set and then filter empty ones out (line 299) rather than add extra logic for every field.

Reply:
I missed that line. 
I think in general we can write something like
```kotlin
limit?.run { queryParams["limit"] = it }
```

Reply:
Ooo, nice! I'll update the other PR.